src/clj/puppetlabs/services/jruby/jruby_metrics_core.clj

```diff
@@ -304,14 +314,19 @@
 (schema/defn jruby-event-callback
   [metrics :- JRubyMetrics
    event :- jruby-schemas/JRubyEvent]
-  (case (:type event)
-    :instance-requested (track-request-instance! metrics event)
-    :instance-borrowed (track-borrow-instance! metrics event)
-    :instance-returned (track-return-instance! metrics event)
-    :lock-requested (track-lock-requested! metrics (:reason event))
-    :lock-acquired (track-lock-acquired! metrics (:reason event))
-    :lock-released (track-lock-released! metrics (:reason event))
-
+  (if-let [[func & args] (case (:type event)
+                           :instance-requested [track-request-instance! metrics event]
+                           :instance-borrowed [track-borrow-instance! metrics event]
+                           :instance-returned [track-return-instance! metrics event]
+                           :lock-requested [track-lock-requested! metrics (:reason event)]
+                           :lock-acquired [track-lock-acquired! metrics (:reason event)]
+                           :lock-released [track-lock-released! metrics (:reason event)]
+                           nil)]
+    (try
+      (apply func args)
+      (catch Exception e
+        (log/error e (trs "Error ocurred while recording metrics for jruby event type: {0}" (:type event)))
+        (throw e)))
```

Comment:
Re-throwing the exception keeps the same behavior that existed prior to this PR. I suppose we could also swallow the exception here in order to allow the JRuby instance to be returned to the pool. Potential downside is that metrics could start drifting, and a pile-up of uncleared `instance-requested` events could cause `max-queued-requests` to start denying requests event though instances had been returned.